src/classes/ADDR_Account_TDTM.cls

```diff
@@ -108,12 +108,19 @@ public with sharing class ADDR_Account_TDTM extends TDTM_Runnable {
         if (acc1 == null) 
            return false;
         
-        return (
+        boolean isSpecified = 
            acc1.BillingStreet != null ||
            acc1.BillingCity != null ||
            acc1.BillingState != null ||
            acc1.BillingPostalCode != null ||
-           acc1.BillingCountry != null);
+            // note that we decided to ignore country so that a default value won't create unnecessary address objects
+           (acc1.BillingCountry != null && !ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled);
+           
```

Comment:
In this second comparison, we want to see if there is a state code, when the picklists are enabled, even if nothing else is specified (or even if country was set automatically)? The way this second condition is written, separate from the others, makes the logic a little hard to understand. Why do we need to check it only if nothing else is set? Can we make it part of the first condition? Also, I think the same comment is there twice.